File: module_base/src/main/java/com/example/base/BaseApplication.java
package com.example.base;

import android.app.Application;
import android.content.Context;
import android.content.Intent;


import java.util.LinkedList;



public abstract class BaseApplication extends Application {
    static private BaseApplication mApplication;

    @Override
    public void onCreate() {
        super.onCreate();
        mApplication = this;
    }



}

File: module_base/src/main/java/com/example/base/base/BaseApplication.kt
package com.example.base.base;

import android.app.Application;
import android.content.Context;





open class  BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        baseApplication = this;
    }


    companion object{
        private lateinit var baseApplication: BaseApplication

        fun getContext(): Context {
            return baseApplication
        }
    }


}
